visualization/app/codeCharta/ui/nodeContextMenu/nodeContextMenu.po.ts

```diff
@@ -10,7 +10,7 @@ export class NodeContextMenuPageObject {
 		await page.waitForSelector("#loading-gif-map", { visible: false })
 	}
 
-	public async isOpen(){
-		return !!await page.waitForSelector("node-context-menu-component", {visible: true})
+	public async isClosed() {
+		return await page.waitForSelector("node-context-menu-component", { visible: false })
```

Comment:
In an async function, it's best not to await the return value if it's not inside a `try` clause. Otherwise we'll delay the return value by another tick without any further benefit (every return value is automatically wrapped in a promise when returned from an async function).

Reply:
What would the best way here then?

We could do something like this 

```ts
public async isClosed() {
	await page.waitForSelector("node-context-menu-component", { visible: false })
        return true
}
```

If the nodeContextMenu would still be open, we'd get a timeout.

Reply:
or as shown [here](https://github.com/puppeteer/puppeteer/issues/1149#issuecomment-434302298)

```ts
public async isClosed() {
	try {
		await page.waitForSelector("node-context-menu-component", { visible: false })
		return true
	} catch (e) {
		return false
	}
}
```

Reply:
It's much simpler:

```suggestion
		return page.waitForSelector("node-context-menu-component", { visible: false })
```

The promise is resolved on the receiving end.